graph/ui-tooltips/src/lib/task-inputs-accordions.tsx

```diff
@@ -0,0 +1,155 @@
+  console.log('TaskInputsListAccordions', projectName);
+  const [filteredInputs, setFilteredInputs] = useState(inputs);
+
+  const resetTextFilter = useCallback(() => {
+    setFilteredInputs(inputs);
+  }, [inputs]);
+
+  const updateTextFilter = useCallback(
+    (textFilter: string) => {
+      if (!inputs || Object.keys(inputs).length === 0) {
+        setFilteredInputs({});
+        return;
+      }
+      if (!textFilter || textFilter.length === 0) {
+        setFilteredInputs(inputs);
+        return;
+      }
+      const filteredInputs: Record<string, ExpandedInputs> = {};
+      Object.keys(inputs).forEach((namedInput) => {
+        Object.keys(inputs[namedInput]).forEach((key) => {
+          Object.keys(inputs[namedInput][key]).forEach((plan) => {
+            const filteredFiles = inputs[namedInput][key][plan].filter(
+              (file) => {
+                return new RegExp(textFilter, 'i').test(file);
+              }
+            );
+            if (filteredFiles.length) {
+              if (!filteredInputs[namedInput]) {
+                filteredInputs[namedInput] = {};
+              }
+              if (!filteredInputs[namedInput][key]) {
+                filteredInputs[namedInput][key] = {};
+              }
+              filteredInputs[namedInput][key][plan] = filteredFiles;
+            }
+          });
+        });
+      });
+      setFilteredInputs(filteredInputs);
+    },
+    [inputs]
+  );
+
+  useEffect(() => {
+    setFilteredInputs(inputs);
+  }, [inputs]);
+
+  if (isLoading) {
+    return (
+      <div className="w-full flex items-center justify-center">
+        <div>Loading...</div>
+      </div>
+    );
+  }
+  if (!inputs || Object.keys(inputs).length === 0) return null;
+
+  return (
+    <>
+      <div className="pb-3 px-3" data-cy="inputs-accordions">
+        <DebouncedTextInput
+          resetTextFilter={resetTextFilter}
+          updateTextFilter={updateTextFilter}
+          initialText={''}
+          placeholderText={'Search files'}
+          Icon={MagnifyingGlassIcon}
+        ></DebouncedTextInput>
+      </div>
+      {filteredInputs && Object.keys(filteredInputs).length > 0 ? (
+        Object.entries(filteredInputs || {}).map(([key, files]) => (
+          <TaskInputsListAccordion
+            heading={key}
+            key={key}
+            containsOnlyOneSection={Object.keys(filteredInputs).length === 1}
+          >
+            <TaskInputsList
+              projectName={projectName}
+              inputs={files}
+              inputName={key}
+            />
+          </TaskInputsListAccordion>
+        ))
+      ) : (
+        <div className="flex justify-center items-center w-full px-4 py-2">
+          No matching files found.
+        </div>
+      )}
+    </>
+  );
+}
+
+export function TaskInputsListAccordion({
+  children,
+  heading,
+  containsOnlyOneSection,
+}: {
+  children: ReactNode;
+  heading: string;
+  containsOnlyOneSection: boolean;
```

Comment:
feels a little weird that you have to pass this into the component. 
Couldn't `TaskInputsListAccordion` just receive the `projectName` & `inputs` and render the lists itself?
You're way better at react so maybe that's just me not being in tune with the way things are done...